src/services/shortLink/test/index.test.ts

```diff
@@ -0,0 +1,552 @@
+
+        const response = await injectPatch(app, MOCK_ALIAS, { item_id: MOCK_ITEM_ID });
+        expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+      });
+    });
+  });
+
+  describe('DELETE /short-links/:itemId', () => {
+    beforeEach(async () => {
+      ({ app, actor } = await build());
+      bob = await saveMember(BOB);
+    });
+
+    it('Not found if delete short links with invalid item id', async () => {
+      const response = await injectDelete(app, MOCK_ALIAS);
+      expect(response.statusCode).toEqual(StatusCodes.NOT_FOUND);
+    });
+
+    describe('Unauthorized and forbidden', () => {
+      afterEach(async () => {
+        const check = await injectGet(app, MOCK_ALIAS);
+        expect(check.statusCode).toEqual(StatusCodes.OK);
+      });
+      it('Unauthorized if delete when not connected', async () => {
+        const { item } = await mockPostData({ author: actor });
+        const payload = mockShortLinkPayload({ item_id: item.id });
+        const post = await injectPost(app, payload);
+        expect(post.statusCode).toEqual(StatusCodes.OK);
+        expect(post.json().alias).toEqual(MOCK_ALIAS);
+
+        logOut(app);
+
+        const response = await injectDelete(app, MOCK_ALIAS);
+        expect(response.statusCode).toEqual(StatusCodes.UNAUTHORIZED);
+      });
+
+      it('Forbidden if delete when non admin user', async () => {
+        const { item } = await mockPostData({
+          author: actor,
+          member: bob,
+          permission: PermissionLevel.Write,
+        });
+        const payload = mockShortLinkPayload({ item_id: item.id });
+        const post = await injectPost(app, payload);
+        expect(post.statusCode).toEqual(StatusCodes.OK);
+        expect(post.json().alias).toEqual(MOCK_ALIAS);
+
+        // log in as bob
+        ({ app } = await build({ member: bob }));
+
+        const response = await injectDelete(app, MOCK_ALIAS);
+        expect(response.statusCode).toEqual(StatusCodes.FORBIDDEN);
+      });
+    });
+
+    describe('Success', () => {
+      afterEach(async () => {
+        const check = await injectGet(app, MOCK_ALIAS);
+        expect(check.statusCode).toEqual(StatusCodes.NOT_FOUND);
+      });
+      it('Success when admin', async () => {
+        const { item } = await mockPostData({
+          author: bob,
+          member: actor,
+          permission: PermissionLevel.Admin,
+        });
+        const payload = mockShortLinkPayload({ item_id: item.id });
+        const post = await injectPost(app, payload);
+        expect(post.statusCode).toEqual(StatusCodes.OK);
+        expect(post.json().alias).toEqual(MOCK_ALIAS);
+
+        ({ app } = await build({ member: bob }));
+
+        const response = await injectDelete(app, MOCK_ALIAS);
+        expect(response.statusCode).toEqual(StatusCodes.OK);
+        expect(response.json().alias).toEqual(MOCK_ALIAS);
+      });
+
+      it('Success when author', async () => {
```

Comment:
Actually this works because when you create an item you get admin writes over it. So actually the only case to test is when you have admin on an item, as you can  loose your admin write but still be the creator.

Reply:
You're right, I will remove this duplicated test.